### PR TITLE
Filter grads

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- remove regularization gradient for variables that don't have prediction gradient

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by Ghostvv** [comment]
should be merged after https://github.com/RasaHQ/rasa/pull/5636

File: rasa/utils/tensorflow/models.py [hunk 1]
```diff
@@ -187,10 +187,31 @@ def train_on_batch(
...
-        with tf.GradientTape() as tape:
-            total_loss = self._total_batch_loss(batch_in)
+        with tf.GradientTape(persistent=True) as tape:
+            prediction_loss = self.batch_loss(batch_in)
+            regularization_loss = tf.math.add_n(self.losses)
+            total_loss = prediction_loss + regularization_loss
+
+        self.total_loss.update_state(total_loss)
+
+        prediction_gradients = tape.gradient(prediction_loss, self.trainable_variables)
```
Comment (tabergma) [inline rasa/utils/tensorflow/models.py:197]: Can we add a couple of more comments? What is this line doing?

File: rasa/utils/tensorflow/models.py [hunk 2]
```diff
@@ -187,10 +187,31 @@ def train_on_batch(
...
+        regularization_gradients = tape.gradient(
+            regularization_loss, self.trainable_variables
+        )
+        del tape
```
Comment (tabergma) [inline rasa/utils/tensorflow/models.py:201]: Why do we delete it?

Comment (Ghostvv) [reply rasa/utils/tensorflow/models.py:201]: that's what they say in tf docs, because we create it with `persistent=True`

**Comment by Ghostvv** [comment]
I fixed dense input dropout here, based on the experiments, it's not a good idea to drop input in dense embedding, however dropping dense input on sparse is good, because there is learnable layer in between

File: rasa/utils/tensorflow/models.py (hunk 1 above)
Comment (Ghostvv) [reply rasa/utils/tensorflow/models.py:199]: added comment

File: rasa/utils/tensorflow/models.py (hunk 2 above)
Comment (Ghostvv) [reply rasa/utils/tensorflow/models.py:206]: added comment
